### ISSUE TITLE
Security Vulnerability

### ISSUE THREAD

**Issue by mschop** [projectsend/projectsend#547]
Hi @ignacionelson ,

I have tried to contact you through contact@projectsend.org but unfortunately no one is answering to me. You have critical security vulnerabilities in your code. Could you please tell me, how you plan to approach this issue or take any other action? Otherwise I have to publish the security problem here on Gitlab.

Best Regards
mschop

**Comment by ignacionelson** [comment]
Hello! Can you please resend you email? Are this vulnerabilities still present on version r1053? Thank you!

**Comment by mschop** [comment]
Hi, I've just sent a new email. The problem was reproducible in master branch last week. NP

**Comment by mschop** [comment]
CVE-2018-10826

**Comment by mschop** [comment]
You can't see the CVE, because it's currently not published (prereserved). As soon as you fixed the security vulnerabilities I will publish it. Please don't wait too long with the fixes ;-). Normally I would give 60 days before I publish the security vulnerability.

I resend the emails to this address.

**Comment by ignacionelson** [comment]
Hello @micschk !! Thanks for your contribution!
I have been exchanging several emails with @mschop and a fix is already being worked on.
However PS is in the middle of a big refactoring stage and the purposed fix will be better applied at a later stage, when an appropiate container is set on the app.
Sorry that it's taking so long but the task at hand is big. I'm really enjoying the process though :)

**Comment by ignacionelson** [comment]
Also, I can honestly say that the information @mschop has been sending me turned out to be invaluable. Concepts I've never heard of that I've learned thanks to this exchange and that I'm waiting the chance to apply to PS. But as I stated before, there's a lot of code that needs to be refactored before that stage!

**Comment by mschop** [comment]
@micschk As @ignacionelson already mentioned, we discussed the issues and how to fix them via email.

CVE's are not intended to put pressure on a producer but to identify every security vulnerability. "Responsible Disclosure" is the ideal way to handle security vulnerabilities. Therefore I did not disclose the details of the vulnerability but kept the communication secret with @ignacionelson .

Ideal way is:
- Create a CVE-Id
- Contact software producer / developer
- Give 30-90 days to fix the issue
- make the details public
